src/import.ts

```diff
@@ -0,0 +1,200 @@
+import { sep } from "path";
+import { Constans } from "./constants";
+import { CustomizeManifest } from "./index";
+import KintoneApiClient, { AuthenticationError } from "./KintoneApiClient";
+import { Lang } from "./lang";
+import { getBoundMessage } from "./messages";
+import { wait } from "./util";
+
+export interface Option {
+  lang: Lang;
+  proxy: string;
+  guestSpaceId: number;
+  destDir: string;
+}
+
+export interface ImportCustomizeManifest {
+  app: string;
+}
+
+interface UploadedFile {
+  type: "FILE";
+  file: {
+    fileKey: string;
+    name: string;
+  };
+}
+
+interface CDNFile {
+  type: "URL";
+  url: string;
+}
+type CustomizeFile = UploadedFile | CDNFile;
+
+interface GetAppCustomizeResp {
+  scope: "ALL" | "ADMIN" | "NONE";
+  desktop: any;
+  mobile: any;
+}
+
+export async function importCustomizeSetting(
+  kintoneApiClient: KintoneApiClient,
+  manifest: ImportCustomizeManifest,
+  status: {
+    retryCount: number;
+  },
+  options: Option
+): Promise<void> {
+  const m = getBoundMessage(options.lang);
+  const appId = manifest.app;
+  let { retryCount } = status;
+
+  try {
+    const appCustomize = kintoneApiClient.getAppCustomize(appId);
+    return appCustomize
+      .then((resp: GetAppCustomizeResp) => {
+        return exportAsManifestFile(appId, options.destDir, resp);
+      })
+      .then((resp: GetAppCustomizeResp) => {
+        downloadCustomizeFiles(kintoneApiClient, appId, options.destDir, resp);
+      })
+      .catch(e => {
+        throw e;
+      });
+  } catch (e) {
+    const isAuthenticationError = e instanceof AuthenticationError;
+    retryCount++;
+    if (isAuthenticationError) {
+      throw new Error(m("E_Authentication"));
+    } else if (retryCount < Constans.MAX_RETRY_COUNT) {
+      await wait(1000);
+      console.log(m("E_Retry"));
+      await importCustomizeSetting(
+        kintoneApiClient,
+        manifest,
+        { retryCount },
+        options
+      );
+    } else {
+      throw e;
+    }
+  }
+}
+
+function exportAsManifestFile(
+  appId: string,
+  destRootDir: string,
+  resp: GetAppCustomizeResp
+): GetAppCustomizeResp {
+  const toNameOrUrl = (destDir: string) => (f: CustomizeFile) => {
+    if (f.type === "FILE") {
+      return `${destDir}${sep}${f.file.name}`;
+    } else {
+      return f.url;
+    }
+  };
+
+  const desktopJs: CustomizeFile[] = resp.desktop.js;
+  const desktopCss: CustomizeFile[] = resp.desktop.css;
+  const mobileJs: CustomizeFile[] = resp.mobile.js;
+
+  const customizeJson: CustomizeManifest = {
+    app: appId,
+    scope: resp.scope,
+    desktop: {
+      js: desktopJs.map(toNameOrUrl(`${destRootDir}${sep}desktop${sep}js`)),
+      css: desktopCss.map(toNameOrUrl(`${destRootDir}${sep}desktop${sep}css`))
+    },
+    mobile: {
+      js: mobileJs.map(toNameOrUrl(`${destRootDir}${sep}mobile${sep}js`))
+    }
+  };
+
+  if (!fs.existsSync(`${destRootDir}`)) {
+    mkdirp.sync(`${destRootDir}`);
+  }
+  fs.writeFile(
```

Comment:
Should it be replaced with `fs.writeSync`?
Currently, `exportAsManifestFile` returns `resp` if this `fs.writeFile` is failed because `fs.writeFile` is async.